content/1-reactivity/1-declare-state/dlight/Name.view.js

```diff
@@ -0,0 +1,12 @@
+import { View } from '@dlightjs/dlight';
+
+@View
+class Name {
+  name = 'John';
+
+  Body() {
+    h1(`Hello ${this.name}`)
+  }
+}
+
+export default Name;
```

Comment:
By curiosity, why did you choose to `export default` like this instead of `export default class Name {` ?

Reply:
In dlight, we use class to declare a component and use `function call` to use one:
```js
@View
class Comp {}

@View
class App {
  Body() {
    Comp()
  }
}
```
This is fine in js. But in ts, this'll arise syntax level errors(even though it's working, but it's ugly in your editor). To we need to force type casting to a self-returning function, which'll resolve this error and give you props auto-completion. But in ts you can't `export class {} as xxx`. So we do it like:
```ts
// ~> Comp.view.ts
@View
class Comp {}
export default Comp as Pretty as Typed<Props>
// ~ App.view.ts
import Comp from "./Comp.view"
@View
class App {
  Body() {
    Comp()
  }
}
```
And here `Pretty` is just an alias of "any" to make the typing pretty. I mean make it "as pretty as" your type...